include/exadg/structure/time_integration/driver_steady_problems.cpp

```diff
@@ -124,8 +124,8 @@ DriverSteady<dim, Number>::do_solve()
     // calculate right-hand side vector
     pde_operator->compute_rhs_linear(rhs_vector, 0.0 /* time */);
 
-    unsigned int const N_iter_linear =
-      pde_operator->solve_linear(solution, rhs_vector, 0.0 /* no mass term */, 0.0 /* time */);
+    unsigned int const N_iter_linear = pde_operator->solve_linear(
+      solution, rhs_vector, 0.0 /* no mass term */, 0.0 /* time */, param.update_preconditioner);
```

Comment:
We have to rethink this critically. In my opinion, this means we initialize the multigrid preconditioner (or, better, the smoothers, coarse grid solver) twice or a steady linear problem: once in the setup-routines, and once within the solver call. However, this is unnecessary work. So we should check if this is the case, and if yes, pass `false` to this function here with a comment that the preconditioner has already been initialized.

Note that this is different for the Newton solver above, because there the preconditioner can be updated every n-th Newton iteration.

Reply:
well actually this evaluates to false here (steady, linear case). so I just put false. `param.update_preconditioner` seemed like here we have some control over that which we do not ...